plugins/GSdx/Renderers/DX11/GSRendererDX11.cpp

```diff
@@ -501,6 +501,114 @@ void GSRendererDX11::EmulateChannelShuffle(GSTexture** rt, const GSTextureCache:
 	}
 }
 
+void GSRendererDX11::EmulateBlending()
+{
+	// Partial port of OGL SW blending. Currently only works for accumulation blend or when
+	// m_sw_blending >= ACC_BLEND_HIGH_D3D11 since it has not been fully tested. There is no check
+	// currently for primitive overlap like in OGL but it should give better results for post-processing effects
+	// that are done by non-overlapping sprites.
+	const GIFRegALPHA& ALPHA = m_context->ALPHA;
+	bool sw_blending         = false;
+
+	m_om_bsel.abe = PRIM->ABE || (PRIM->AA1 && m_vt.m_primclass == GS_LINE_CLASS);
+	m_om_bsel.a = ALPHA.A;
+	m_om_bsel.b = ALPHA.B;
+	m_om_bsel.c = ALPHA.C;
+	m_om_bsel.d = ALPHA.D;
+
+	if (m_env.PABE.PABE)
+	{
+		if (m_om_bsel.a == 0 && m_om_bsel.b == 1 && m_om_bsel.c == 0 && m_om_bsel.d == 1)
+		{
+			// this works because with PABE alpha blending is on when alpha >= 0x80, but since the pixel shader
+			// cannot output anything over 0x80 (== 1.0) blending with 0x80 or turning it off gives the same result
+
+			m_om_bsel.abe = 0;
+		}
+		else
+		{
+			//Breath of Fire Dragon Quarter triggers this in battles. Graphics are fine though.
+			//ASSERT(0);
+		}
+	}
+
+	// Accumulation blend
+	uint8 blend_index  = uint8(((ALPHA.A * 3 + ALPHA.B) * 3 + ALPHA.C) * 3 + ALPHA.D);
+	int blend_flag = m_dev->GetBlendFlags(blend_index);
+
+	bool impossible_or_free_blend =
+		(blend_flag & (BLEND_A_MAX|BLEND_NO_BAR|BLEND_ACCU))
+		|| m_bind_rtsample;
+
+	bool accumulation_blend = !!(blend_flag & BLEND_ACCU);
+
+	switch (m_sw_blending)
+	{
+	case ACC_BLEND_HIGH_D3D11:
+		if (!m_vt.m_alpha.valid && (ALPHA.C == 0)) GetAlphaMinMax();
+		sw_blending |= (ALPHA.A != ALPHA.B) &&
+			((ALPHA.C == 0 && m_vt.m_alpha.max > 128) || (ALPHA.C == 2 && ALPHA.FIX > 128u));
+		sw_blending |= (ALPHA.A != ALPHA.B) && (ALPHA.C == 1);
+		// fall through
+	case ACC_BLEND_MEDIUM_D3D11:
+	case ACC_BLEND_BASIC_D3D11:
+		sw_blending |= impossible_or_free_blend;
+	default: break;
+	}
+
+	// Color clip
+	if (m_env.COLCLAMP.CLAMP == 0)
+	{
+		if (accumulation_blend)
+		{
+			m_ps_sel.hdr = 1;
+			sw_blending = true;
+		}
+		else if (sw_blending)
+		{
+			m_ps_sel.colclip = 1;
+		}
+		else
+		{
+			m_ps_sel.hdr = 1;
+		}
+	}
+
+	if (sw_blending)
+	{
+		m_ps_sel.blend_a = ALPHA.A;
+		m_ps_sel.blend_b = ALPHA.B;
+		m_ps_sel.blend_c = ALPHA.C;
+		m_ps_sel.blend_d = ALPHA.D;
+
+		if (accumulation_blend)
+		{
+			m_om_bsel.accu_blend = 1;
+
+			if (ALPHA.A == 2) {
+				// The blend unit does a reverse subtraction so it means
+				// the shader must output a positive value.
+				// Replace 0 - Cs by Cs - 0
+				m_ps_sel.blend_a = ALPHA.B;
+				m_ps_sel.blend_b = 2;
+			}
+			// Remove the addition/substraction from the SW blending
+			m_ps_sel.blend_d = 2;
+		}
+		else
+		{
+			// Disable HW blending
+			m_om_bsel.abe = false;
+
+			m_bind_rtsample |= !(blend_flag & BLEND_NO_BAR);
+		}
+
+		// Require the fix alpha vlaue
+		if (ALPHA.C == 2)
+			ps_cb.Af.x = (float)ALPHA.FIX / 128.0f;
+	}
```

Comment:
I think we can also move clr blending here like on gl when there is no sw_blending, handle it the same on gl and remove the IsCLR1 bool in blend selector struct.

Reply:
Actually, even this might be a good change to do now because really its part of the blend map changes and using the blend flags on the DX11 side. It also cleans up the code a bit and might be forgotten later if we leave it.

Reply:
You need to remove this in gsdevice11.h since it will be useless with these new changes
```
		bool IsCLR1() const
		{
			return (key & 0x19f) == 0x93; // abe == 1 && a == 1 && b == 2 && d == 1
		}
```